android/filament-utils-android/src/main/cpp/Utils.cpp

```diff
@@ -102,25 +102,20 @@ JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void*) {
     jclass ktxloaderClass = env->FindClass("com/google/android/filament/utils/KTXLoader");
     if (ktxloaderClass == nullptr) return JNI_ERR;
     static const JNINativeMethod ktxMethods[] = {
-        {strdup("nCreateKTXTexture"), strdup("(JLjava/nio/Buffer;IZ)J"), reinterpret_cast<void*>(nCreateKTXTexture)},
-        {strdup("nCreateIndirectLight"), strdup("(JLjava/nio/Buffer;IZ)J"), reinterpret_cast<void*>(nCreateIndirectLight)},
-        {strdup("nCreateSkybox"), strdup("(JLjava/nio/Buffer;IZ)J"), reinterpret_cast<void*>(nCreateSkybox)},
-        {strdup("nGetSphericalHarmonics"), strdup("(Ljava/nio/Buffer;I[F)Z"), reinterpret_cast<void*>(nGetSphericalHarmonics)},
+        {(char*)"nCreateKTXTexture", (char*)"(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateKTXTexture)},
+        {(char*)"nCreateIndirectLight", (char*)"(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateIndirectLight)},
+        {(char*)"nCreateSkybox", (char*)"(JLjava/nio/Buffer;IZ)J", reinterpret_cast<void*>(nCreateSkybox)},
+        {(char*)"nGetSphericalHarmonics", (char*)"(Ljava/nio/Buffer;I[F)Z", reinterpret_cast<void*>(nGetSphericalHarmonics)},
     };
     rc = env->RegisterNatives(ktxloaderClass, ktxMethods, sizeof(ktxMethods) / sizeof(JNINativeMethod));
-    free(ktxMethods[0].name); free(ktxMethods[0].signature);
-    free(ktxMethods[1].name); free(ktxMethods[1].signature);
-    free(ktxMethods[2].name); free(ktxMethods[2].signature);
-    free(ktxMethods[3].name); free(ktxMethods[3].signature);
     if (rc != JNI_OK) return rc;
 
     // HDRLoader
     jclass hdrloaderClass = env->FindClass("com/google/android/filament/utils/HDRLoader");
     if (hdrloaderClass == nullptr) return JNI_ERR;
     static const JNINativeMethod hdrMethods[] = {
-        {strdup("nCreateHDRTexture"), strdup("(JLjava/nio/Buffer;II)J"), reinterpret_cast<void*>(nCreateHDRTexture)},
+        {(char*)"nCreateHDRTexture", (char*)"(JLjava/nio/Buffer;II)J", reinterpret_cast<void*>(nCreateHDRTexture)},
```

Comment:
We could also use an std::string. I'm surprised this works without a const_cast.

Reply:
It's clearly an oversight. Let's just cast to `char*`.

Reply:
`const_cast` would be cleaner -- but the basic C cast always wins (it's stronger than `reinterpret_cast`). I think this is fine. 
I'm surprised that the `reinterpret_cast` for the function pointer is needed actually, any pointer should automatically cast to void. oh well.